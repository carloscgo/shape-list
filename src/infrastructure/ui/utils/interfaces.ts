import { DragEvent, ReactNode } from 'react'
import { Some } from '../../../application'
import { IdShape, Shape, TypeShape } from '../../../domain/models/Shape'

export type PositionsT = 'left' | 'right'

export type PropsShape = {
  shape: Shape
  onAdd: (position: PositionsT, index: IdShape) => void
  onDelete: (index: IdShape) => void
  showButton?: boolean
}

export type PropsShapeList = {
  shapes: Shape[]
  onAdd: (position: PositionsT, type: TypeShape, index: IdShape) => void
  onDelete: (index: IdShape) => void
}

export type PropsAddButton = {
  onAdd: () => void
  [key: string]: Some
}

export type PropsDeleteButton = {
  onDelete: () => void
  [key: string]: Some
}

export type DragDropT = DragEvent<HTMLDivElement>

export type PropsModal = {
  color: 'red' | 'indigo'
  title: string
  message?: string
  children?: ReactNode
  labelButton: string
  show: boolean
  onConfirm: () => void
  onClose: (show: boolean) => void
}
