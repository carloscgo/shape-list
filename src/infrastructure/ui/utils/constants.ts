// infrastructure/ui/utils/constants.ts

export const Status = {
  loading: 'loading',
  success: 'success',
  error: 'error',
}

export const Positions = {
  left: 'left',
  right: 'right',
}

export const Store = 'shapes'

export const SizeButton = '50px'

export const SizeIcon = '40px'

export const SizeShape = '200px'

export const SizeContainer = '300px'

// Properties to Modal Component

export const PropsTransitionChild1 = {
  enter: 'ease-out duration-300',
  enterFrom: 'opacity-0',
  enterTo: 'opacity-100',
  leave: 'ease-in duration-200',
  leaveFrom: 'opacity-100',
  leaveTo: 'opacity-0',
}

export const PropsTransitionChild2 = {
  enter: 'ease-out duration-300',
  enterFrom: 'opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95',
  enterTo: 'opacity-100 translate-y-0 sm:scale-100',
  leave: 'ease-in duration-200',
  leaveFrom: 'opacity-100 translate-y-0 sm:scale-100',
  leaveTo: 'opacity-0 translate-y-4 sm:translate-y-0 sm:scale-95',
}

export const classIcon = (color: string) =>
  ({
    red: `h-6 w-6 text-red-600`,
    indigo: `h-6 w-6 text-indigo-600`,
  }[color])

export const classButton = (color: string) =>
  ({
    red: `inline-flex w-full justify-center rounded-md bg-red-600 px-3 py-2 text-sm font-semibold text-white shadow-sm hover:bg-red-500 sm:ml-3 sm:w-auto`,
    indigo: `inline-flex w-full justify-center rounded-md bg-indigo-600 px-3 py-2 text-sm font-semibold text-white shadow-sm hover:bg-indigo-500 sm:ml-3 sm:w-auto`,
  }[color])
