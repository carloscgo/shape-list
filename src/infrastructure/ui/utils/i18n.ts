// infrastructure/ui/utils/i18n.ts

import i18n from 'i18next'
import { useTranslation, initReactI18next } from 'react-i18next'
import enJSON from '../locales/en.json'
import esJSON from '../locales/es.json'
import { getItemStorage } from '../hooks/useLocalStorage'

export const languages = [
  {
    key: 'en',
    name: 'English',
  },
  {
    key: 'es',
    name: 'Espa√±ol',
  },
]
export const key = 'language'
export const fallbackLanguage = languages[0]

export default function config() {
  const language = getItemStorage(key, fallbackLanguage)

  i18n.use(initReactI18next)
  i18n.init({
    resources: {
      en: { ...enJSON },
      es: { ...esJSON },
    },
    lng: language.key,
    debug: false,
  })
}

export { useTranslation, i18n, initReactI18next }
