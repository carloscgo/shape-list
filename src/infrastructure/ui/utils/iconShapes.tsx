import { CgShapeCircle, CgShapeSquare, CgShapeTriangle } from 'react-icons/cg'

const iconsShapes = {
  Circle: (size: string) => (
    <CgShapeCircle size={size} className="text-teal-700" />
  ),
  Square: (size: string) => (
    <CgShapeSquare size={size} className="text-indigo-700" />
  ),
  Triangle: (size: string) => (
    <CgShapeTriangle size={size} className="text-orange-700" />
  ),
}

export default iconsShapes
