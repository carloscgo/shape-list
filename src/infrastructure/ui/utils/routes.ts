// infrastructure/ui/utils/routes.ts

import { BrowserRouter, Routes, Route, Link, useParams } from 'react-router-dom'

const routes = {
  home: '/',
  list: '/',
  new: '/shape/new',
  edit: '/shape/edit',
  details: '/shape/details',
}

export default routes
export { BrowserRouter, Routes, Route, Link, useParams }
