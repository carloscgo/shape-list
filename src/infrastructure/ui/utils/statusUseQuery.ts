const StatusMutation = {
    loading: 'loading',
    success: 'success',
    error: 'error',
    idle: 'idle',
}

export default StatusMutation;