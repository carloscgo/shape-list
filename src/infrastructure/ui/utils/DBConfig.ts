// infrastructure/ui/utils/DBConfig.ts

import { Store } from './constants'

export const DBConfig = {
  name: 'ShapeDB',
  version: 1,
  objectStoresMeta: [
    {
      store: Store,
      storeConfig: { keyPath: 'id', autoIncrement: true },
      storeSchema: [
        { name: 'type', keypath: 'type', options: { unique: false } },
        { name: 'icon', keypath: 'icon', options: { unique: false } },
      ],
    },
  ],
}
