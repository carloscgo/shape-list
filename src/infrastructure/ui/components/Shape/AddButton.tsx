import { memo } from 'react'
import { HiPlusCircle } from 'react-icons/hi2'
import { PropsAddButton } from '../../utils/interfaces'
import { SizeButton, SizeIcon } from '../../utils/constants'

const AddButton = memo(({ onAdd, ...props }: PropsAddButton) => {
  const { className, ...attrs } = props

  return (
    <div
      role="button"
      aria-label="add shape"
      className={`flex items-center justify-center cursor-pointer bg-white rounded-full w-[${SizeButton}] h-[${SizeButton}] mx-auto ${className}`}
      onClick={onAdd}
      {...attrs}
    >
      <HiPlusCircle size={SizeIcon} className="text-blue-600" />
    </div>
  )
})

export default AddButton
