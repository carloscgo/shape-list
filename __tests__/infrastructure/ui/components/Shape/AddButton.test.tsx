import {
  expect,
  describe,
  it,
  render,
  fireEvent,
  React,
} from '../../../../../setupTest'
import { vi } from 'vitest'

import AddButton from '../../../../../src/infrastructure/ui/components/Shape/AddButton'
import { SizeButton } from '../../../../../src/infrastructure/ui/utils/constants'

describe('AddButton', () => {
  const mockOnClick = vi.fn()

  it('should render without crashing', () => {
    const { container } = render(<AddButton onAdd={mockOnClick} />)

    expect(container).toMatchSnapshot()
  })

  it('should invoke the onAdd callback when clicked', () => {
    const { getByRole } = render(<AddButton onAdd={mockOnClick} />)
    const addButton = getByRole('button')
    fireEvent.click(addButton)
    expect(mockOnClick).toHaveBeenCalledTimes(1)
  })

  it('should render with the correct classes', () => {
    const { getByRole } = render(
      <AddButton onAdd={mockOnClick} className="test-class" />
    )
    const addButton = getByRole('button')
    expect(addButton).toHaveClass('bg-white')
    expect(addButton).toHaveClass('rounded-full')
    expect(addButton).toHaveClass(`w-[${SizeButton}]`)
    expect(addButton).toHaveClass(`h-[${SizeButton}]`)
    expect(addButton).toHaveClass('mx-auto')
    expect(addButton).toHaveClass('test-class')
  })

  it('should render without classes', () => {
    const { getByRole } = render(<AddButton onAdd={mockOnClick} />)
    const addButton = getByRole('button')
    expect(addButton).toHaveClass('bg-white')
    expect(addButton).toHaveClass('rounded-full')
    expect(addButton).toHaveClass(`w-[${SizeButton}]`)
    expect(addButton).toHaveClass(`h-[${SizeButton}]`)
    expect(addButton).toHaveClass('mx-auto')
  })

  it('should render with custom attributes', () => {
    const { getByRole } = render(
      <AddButton onAdd={mockOnClick} data-testid="test-id" />
    )
    const addButton = getByRole('button')
    expect(addButton).toHaveAttribute('data-testid', 'test-id')
  })
})
