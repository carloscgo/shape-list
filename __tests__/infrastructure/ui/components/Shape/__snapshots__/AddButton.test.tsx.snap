// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddButton > should render without crashing 1`] = `
<div>
  <div
    aria-label="add shape"
    class="flex items-center justify-center cursor-pointer bg-white rounded-full w-[50px] h-[50px] mx-auto undefined"
    role="button"
  >
    <svg
      aria-hidden="true"
      class="text-blue-600"
      fill="currentColor"
      height="40px"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="40px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12 2.25c-5.385 0-9.75 4.365-9.75 9.75s4.365 9.75 9.75 9.75 9.75-4.365 9.75-9.75S17.385 2.25 12 2.25zM12.75 9a.75.75 0 00-1.5 0v2.25H9a.75.75 0 000 1.5h2.25V15a.75.75 0 001.5 0v-2.25H15a.75.75 0 000-1.5h-2.25V9z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;
